### DISCUSSION TITLE
Provide OnyxSection instead of OnyxHeadline

### DISCUSSION THREAD

**Discussion by JoCa96** [SchwarzIT/onyx#2246] [Ideas]
While [headlines](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/Heading_Elements) are one of the most important elements on websites, they are easy to misuse:
- levels might be skipped
- `h1` might be used multiple times
- content might not be correctly associated with a headline
- might be completely unused

Instead of providing a standalone `OnyxHeadline` component, which just represents a heading without content, we could implement a component API that
- couples headline and content (we are probably [going to need](https://github.com/SchwarzIT/onyx/issues/569) this coupling anyway for the `reveal` and `select` mode)
- supports relative heading levels (`<OnyxHeadline level="child">` instead of `<OnyxHeadline is="h2">`) via Vue `inject` and `provide`

**Comment by larsrickert** [comment]
I get the idea but I am not sure if we should really abstract it this far.
Imo, it would be more confusing/unclear to the user what `level="child"` is instead of `is="h2"`.
Also if components are misused, that should be responsibility of the user or our docs.

Also, coupling the content to it could be very edge case unfriendly since headlines might also be used in components like cards or with different content padding

However, I am still open to try this out and check how it feels :)
